components/flash_memory/flash_nvs.c

```diff
@@ -18,9 +50,97 @@ NVSResult NVS_init(void) {
   }
 
   nvs.handle = 0;
+  
+  length = sizeof(*data_array)/sizeof(data_array[0]);
```

Comment:
To się będzie zawsze równać jeden. Jako argument dostajesz wskaźnik na pierwszy element tablicy NVSData, więc zapis *data_array jest równoważny z data_array[0], więc tu i tu dostaniesz ten sam rozmiar.